src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java

```diff
@@ -149,18 +160,26 @@ public void setupChild(Configuration childConf)
             throws CheckstyleException {
         final String name = childConf.getName();
         final Object module = moduleFactory.createModule(name);
-        if (!(module instanceof AbstractCheck)) {
+        if (module instanceof AbstractCheck || module instanceof TreeWalkerFilter) {
```

Comment:
This is mess in code. Use the same approach as in Checker. It is easier to read and understand.

```java
if (module instanceof AutomaticBean) {
    final AutomaticBean bean = (AutomaticBean) module;
    bean.contextualize(childContext);
    bean.configure(childConf);
}

if (module instanceof AbstractCheck) {
   final AbstractCheck check = (AbstractCheck) module;
   check.init();
   registerCheck(check);
}
else if (module instanceof TreeWalkerFilter) {
    filters.add((TreeWalkerFilter) module);
}
else {
    throw new CheckstyleException(
```